pkgs/by-name/ey/eyewitness/package.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3Packages
+, cmake
+, xvfb-run
+, firefox-esr
+, geckodriver
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
+  pname = "eye-witness";
+  version = "20230525.1";
+
+  src = fetchFromGitHub {
+    owner = "redsiege";
+    repo = "EyeWitness";
+    rev = "v${version}";
+    hash = "sha256-nSPpPbwqagc5EadQ4AHgLhjQ0kDjmbdcwE/PL5FDL4I=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    setuptools
+  ] ++ [
+    makeWrapper
+  ];
+
+  buildInputs = with python3Packages; [
+    selenium
+    fuzzywuzzy
+    pyvirtualdisplay
+    pylev
+    netaddr
+    pydevtool
+  ] ++ [
+    firefox-esr
+    xvfb-run
+    geckodriver
```

Comment:
Just out of curiosity, what are they for?

Reply:
due to the nature of the script, selenium based projects require web browser engine i.e Gecko in this situation the program collects content like HTML, XML, images and GeckoDriver is a connecting link to the Firefox browser for Selenium based projects. It is a proxy that helps you communicate with Gecko-based browsers.